src/auth/auth.ts

```diff
@@ -599,11 +603,126 @@ class BaseAuth {
 }
 
 
+/**
+ * The tenant aware Auth class.
+ */
+export class TenantAwareAuth extends BaseAuth {
+  public readonly tenantId: string;
+
+  /**
+   * The TenantAwareAuth class constructor.
+   *
+   * @param {Auth} auth The Auth instance that created this tenant.
+   * @param tenantId The corresponding tenant ID.
+   * @constructor
+   */
+  constructor(private readonly auth: Auth, tenantId: string) {
```

Comment:
Looks like it makes sense for this constructor to just accept `FirebaseApp` instead of `Auth`.

Reply:
Done